blog/articles/set-up-a-local-knative-environment-with-kind.md

```diff
@@ -0,0 +1,315 @@
+kourier   NodePort   10.98.179.178   <none>        80:31080/TCP,443:31443/TCP   87m
+
+$ docker ps -a
+CONTAINER ID        IMAGE                  COMMAND                  CREATED             STATUS              PORTS                                                                      NAMES
+d53c275d7461        kindest/node:v1.18.2   "/usr/local/bin/entr…"   4 hours ago         Up 4 hours          127.0.0.1:49350->6443/tcp, 0.0.0.0:80->31080/tcp, 0.0.0.0:443->31443/tcp   knative-control-plane
+```
+
+The ports, and how they’re tied to the host, should be the same as you’ve defined in the clusterconfig file. For example, port 31380 in the cluster is exposed as port 80.
+
+## Your first app
+Everyone enjoys a good “hello world” example as the first app you deploy, so let’s do that here too. The cluster, Knative, and the networking components are ready so the last part is to deploy an app. The straightforward [Go app](https://knative.dev/docs/eventing/samples/helloworld/helloworld-go/) that already exists, is an excellent candidate to deploy. The first step there is to create a yaml file with the hello world service definition.
+
+```bash
+cat > service.yaml <<EOF
+apiVersion: serving.knative.dev/v1 # Current version of Knative
+kind: Service
+metadata:
+  name: helloworld-go # The name of the app
+  namespace: default # The namespace the app will use
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-samples/helloworld-go # The URL to the image of the app
+          env:
+            - name: TARGET # The environment variable printed out by the sample app
+              value: "Hello Knative Serving is up and running with Kourier!!"
+EOF
+```
+
+To deploy your app to Knative, run
+
+```bash
+$ kubectl apply --filename service.yaml
+service.serving.knative.dev/helloworld-go created
+```
+
+To validate your deployment, you can use  `kubectl get ksvc`. While your cluster is taking care of the configuration of the different elements that make up the service the output of that command will show that the revision is missing. After some time, you’ll see *ready* change to true.
+
+```bash
+$ kubectl get ksvc
+NAME            URL                                             LATESTCREATED         LATESTREADY   READY     REASON
+helloworld-go   http://helloworld-go.default.127.0.0.1.nip.io   helloworld-go-fqqs6                 Unknown   RevisionMissing
+
+$ kubectl get ksvc
+NAME            URL                                             LATESTCREATED         LATESTREADY           READY   REASON
+helloworld-go   http://helloworld-go.default.127.0.0.1.nip.io   helloworld-go-fqqs6   helloworld-go-fqqs6   True
+```
+
+A running and ready service leaves one final step… checking that the code returns what you expect. To do that, you can send a cURL request to the URL listed above. Because you’ve mapped port 80 of the host to be forwarded to the cluster, and set the DNS, you can use that exact URL.
```

Comment:
```suggestion
The final step is to test your application, by checking that the code returns what you expect. You can do this by sending a cURL request to the URL listed above.

Because this example mapped port 80 of the host to be forwarded to the cluster and set the DNS, you can use the exact URL.
```